planner/core/explain.go

```diff
@@ -238,7 +238,14 @@ func (p *PhysicalHashJoin) ExplainInfo() string {
 	}
 
 	buffer.WriteString(p.JoinType.String())
-	fmt.Fprintf(buffer, ", inner:%s", p.Children()[p.InnerChildIdx].ExplainID())
+	var InnerChildIdx = p.InnerChildIdx
+	if p.OuterHashJoin && ((p.JoinType == LeftOuterJoin && InnerChildIdx == 1) || (p.JoinType == RightOuterJoin && InnerChildIdx == 0)) {
+		InnerChildIdx = 1 - InnerChildIdx
+	}
+	fmt.Fprintf(buffer, ", inner:%s", p.Children()[InnerChildIdx].ExplainID())
+	if p.OuterHashJoin {
+		buffer.WriteString(" (REVERSED)")
```

Comment:
We'd better not to use the `"REVERSED"` to indicate an `OuterHashJoin`. Because that's confusing when it appears to `explain` results. A good way to distinguish them is to change the executing operator name to "InnerHashJoin" and "OuterHashJoin".

Reply:
I think we do not need to introduce "InnerHashJoin" and "OuterHashJoin", it'll be confusing.
Before this commit, we use `inner` to mark the inner side and the side which is used to build the hash table.
Maybe we can add a label `hashtable` to mark the build side after this commit.

Reply:
buildside and probeside? Because `build` and `probe` are two standard words to describe the processing of hash joins.

Reply:
I think `buildside` is ok. Or we can simplify it to `build`

Reply:
> We'd better not to use the `"REVERSED"` to indicate an `OuterHashJoin`. Because that's confusing when it appears to `explain` results. A good way to distinguish them is to change the executing operator name to "InnerHashJoin" and "OuterHashJoin".

using `OuterHashJoin` may be confused with the `left/right outer join`.

Reply:
> I think `buildside` is ok. Or we can simplify it to `build`

This also affect other cases, I don't know which way is ok.

Reply:
related issue:
https://github.com/pingcap/tidb/issues/12985